lms/djangoapps/courseware/access_utils.py

```diff
@@ -0,0 +1,77 @@
+"""
+Simple utility functions for computing access.
+It allows us to share code between access.py and block transformers.
+"""
+
+from datetime import timedelta
+from django.conf import settings
+from django.utils.timezone import now
+from logging import getLogger
+from student.roles import CourseBetaTesterRole
+from courseware.masquerade import is_masquerading_as_student
+from courseware.access_response import AccessResponse, StartDateError
+from xmodule.util.django import get_current_request_hostname
+
+
+DEBUG_ACCESS = False
+log = getLogger(__name__)
+
+ACCESS_GRANTED = AccessResponse(True)
+ACCESS_DENIED = AccessResponse(False)
+
+
+def debug(*args, **kwargs):
+    """
+    Helper function for local debugging.
+    """
+    # to avoid overly verbose output, this is off by default
+    if DEBUG_ACCESS:
+        log.debug(*args, **kwargs)
+
+
+def adjust_start_date(user, days_early_for_beta, start, course_key):
+    """
+    If user is in a beta test group, adjust the start date by the appropriate number of
+    days.
+
+    Returns:
+        A datetime.  Either the same as start, or earlier for beta testers.
+    """
+    if days_early_for_beta is None:
+        # bail early if no beta testing is set up
+        return start
+
+    if CourseBetaTesterRole(course_key).has_user(user):
+        debug("Adjust start time: user in beta role for %s", course_key)
```

Comment:
I didn't plan to change the semantics of this code (at least not intentionally - barring the `now` change).  It was only meant to be a refactor of the existing code so it can be reused by the start_date transformer.
So I'm going to keep the `debug` indirection as is.  Someone must have introduced it since it was too noisy.